<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jan Lanzer" />


<title>Cluster.identification</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SFB_arrythmia</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/JanLanzer/SFB_arrythmia">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Cluster.identification</h1>
<h4 class="author">Jan Lanzer</h4>
<h4 class="date">7/27/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>SFB_arrythmia/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210720code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210720)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210720code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210720)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomJanLanzerSFBarrythmiatree9fc2568bd22572ff36372d38997691d681abc29atargetblank9fc2568a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/JanLanzer/SFB_arrythmia/tree/9fc2568bd22572ff36372d38997691d681abc29a" target="_blank">9fc2568</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomJanLanzerSFBarrythmiatree9fc2568bd22572ff36372d38997691d681abc29atargetblank9fc2568a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/JanLanzer/SFB_arrythmia/tree/9fc2568bd22572ff36372d38997691d681abc29a" target="_blank">9fc2568</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/data.limma.rds
    Ignored:    data/data.limma_filt.rds
    Ignored:    data/gene_id_symbol_dictionary_July2021.rds
    Ignored:    data/raw.combined.rds

Untracked files:
    Untracked:  analysis/normalize.R
    Untracked:  analysis/normalize_test.R
    Untracked:  analysis/process.R
    Untracked:  analysis/utils_preprocess.R
    Untracked:  data/genes of interest 2.txt
    Untracked:  data/genes_felix.csv
    Untracked:  data/raw/
    Untracked:  omnipathr-log/
    Untracked:  output/DEA_AF1_AF2.csv
    Untracked:  output/ETV_regs.xls
    Untracked:  output/KCNK3_potentialTFs.csv
    Untracked:  output/PC1_loadings_outlier.csv

Unstaged changes:
    Modified:   analysis/funcomics.Rmd
    Modified:   analysis/funcomics_filtered.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/cluster.Rmd</code>) and HTML (<code>docs/cluster.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/9fc2568bd22572ff36372d38997691d681abc29a/analysis/cluster.Rmd" target="_blank">9fc2568</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-08-03
</td>
<td>
wflow_publish(“analysis/cluster.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/JanLanzer/SFB_arrythmia/787dca5cbf96cade4a217e135c73092c6253b723/docs/cluster.html" target="_blank">787dca5</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/df335afce6a2dc3591785f706ddb1753cd837212/analysis/cluster.Rmd" target="_blank">df335af</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
<td>
wflow_publish(“analysis/cluster.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/JanLanzer/SFB_arrythmia/4bfadb4d9fa2630d570681a23dbc542451009603/docs/cluster.html" target="_blank">4bfadb4</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/55da40ace70824666e107dfc5bda3823d2c4e978/analysis/cluster.Rmd" target="_blank">55da40a</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
<td>
wflow_publish(“analysis/cluster.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="patient-distance-and-filtering-of-outliers" class="section level1">
<h1>patient distance and filtering of outliers</h1>
<pre class="r"><code>library(lsa)

#use cosine similarity on all GEX, all samples:
voom_count= data.all$GEX$voom_count
ds. = cosine(voom_count)
pheatmap(ds.)</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/4bfadb4d9fa2630d570681a23dbc542451009603/docs/figure/cluster.Rmd/unnamed-chunk-1-1.png" target="_blank">4bfadb4</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#plot clustering on all GEX, outliers removed:
voom_count= data.filt$GEX$voom_count
ds. = cosine(voom_count)
pheatmap(ds.)</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-2">
Past versions of unnamed-chunk-1-2.png
</button>
</p>
<div id="fig-unnamed-chunk-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/4bfadb4d9fa2630d570681a23dbc542451009603/docs/figure/cluster.Rmd/unnamed-chunk-1-2.png" target="_blank">4bfadb4</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>af.patients= data.filt$target %&gt;% filter(group== &quot;AF&quot;) %&gt;% pull(sampleID)
ds. = cosine(voom_count[,af.patients])
pheatmap(ds.)</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># #eucledian:
# ds. = dist(t(voom_count))
# pheatmap(ds.)
# ds.</code></pre>
</div>
<div id="find-cluster-based-on-gene-sets" class="section level1">
<h1>Find cluster based on gene sets</h1>
<p>1 use prior knowldege gene set 2 use NABA (fibrosis)</p>
<pre class="r"><code>#load gene set with AF prior knowledge: 
genes_felix = read.csv(&quot;data/genes_felix.csv&quot;)[,1]
prior.k= enframe(genes_felix) %&gt;%
  mutate(mor = 1, likelihood= 1, gene.set= &quot;prior&quot;)

  
#load  NABA
processNABA = function(filepath = &quot;/home/jan/R-projects/sc-exploration/data/NABAgsets.xls&quot;) {
  con = file(filepath, &quot;r&quot;)
  naba_gsets = list()
  while ( TRUE ) {
    line = readLines(con, n = 1)
    if ( length(line) == 0 ) {
      break
    }
    split_line = unlist(strsplit(line,split=&quot;\t&quot;))
    naba_gsets[[split_line[1]]] = split_line[3:length(split_line)]
  }
  close(con)
  return(naba_gsets)
}

naba_gsets= processNABA()

x = map(names(naba_gsets) , function(y){
  enframe(naba_gsets[[y]]) %&gt;% mutate(gene.set= y)
  })

naba_gsets2= do.call(rbind, x) %&gt;%
  mutate(mor = 1, likelihood= 1)

all_genesets= rbind(naba_gsets2, prior.k)

#enrich_ 

statistics = c(&quot;gsva&quot;, &quot;scira&quot;)
.R= decouple(data.filt$GEX$voom_count, network = all_genesets, .source = gene.set, .target = value, statistics = statistics, include_time = F) </code></pre>
<pre><code>Estimating GSVA scores for 11 gene sets.
Estimating ECDFs with Gaussian kernels

  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |======                                                                |   9%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |================================                                      |  45%
  |                                                                            
  |======================================                                |  55%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |================================================================      |  91%
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre class="r"><code>#plot contrast AF-CT 
.R %&gt;% 
  left_join(data.all$target %&gt;% rename(condition= sampleID)) %&gt;% 
  ggplot(.,aes(x=tf, y= score, fill = group))+
  facet_grid(rows= &quot;statistic&quot;)+
  geom_boxplot()+
  theme(axis.text.x = element_text(angle= 60, hjust= 1
  ))</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/4bfadb4d9fa2630d570681a23dbc542451009603/docs/figure/cluster.Rmd/unnamed-chunk-2-1.png" target="_blank">4bfadb4</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Identifying AF clusters with NABA scores
map(statistics, function(x){
  AF.NABA= .R %&gt;%
    filter(statistic== x)%&gt;%
    left_join(data.all$target %&gt;% rename(condition= sampleID)) %&gt;%
    filter(group == &quot;AF&quot;) %&gt;% 
    select(condition, score, tf) %&gt;%
    pivot_wider( names_from= condition, values_from = score) %&gt;% column_to_rownames(&quot;tf&quot;)
  
  pheatmap(AF.NABA)
  
})</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-2">
Past versions of unnamed-chunk-2-2.png
</button>
</p>
<div id="fig-unnamed-chunk-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/4bfadb4d9fa2630d570681a23dbc542451009603/docs/figure/cluster.Rmd/unnamed-chunk-2-2.png" target="_blank">4bfadb4</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cluster.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]

[[2]]</code></pre>
<p>samples 1.28, 1-29, 1.26 and 1.27 show high fibrosis and low gene expression of the submitted prior</p>
</div>
<div id="now-we-will-compare-actual-gene-expression-values-for-the-genes-of-interest" class="section level1">
<h1>now we will compare actual gene expression values for the genes of interest</h1>
<pre class="r"><code>AF.patients= data.filt$target %&gt;% filter(group== &quot;AF&quot;) %&gt;% pull(sampleID)
all.patients=data.filt$target$sampleID 
cal_z_score &lt;- function(x){
  (x - mean(x)) / sd(x)
}
 
plot.heatmap.for.genes= function(gene, patients){
  x= voom_count[grepl(gene, rownames(voom_count)),patients]
  data_subset_norm &lt;- t(apply(x, 1, cal_z_score))

  pheatmap(data_subset_norm)
  pheatmap(x)
}

plot.heatmap.for.genes(&quot;KCNK&quot;, all.patients)</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/4bfadb4d9fa2630d570681a23dbc542451009603/docs/figure/cluster.Rmd/unnamed-chunk-3-1.png" target="_blank">4bfadb4</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cluster.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.heatmap.for.genes(&quot;KCNK&quot;, AF.patients)</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/cluster.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.heatmap.for.genes(&quot;ETV&quot;, all.patients)</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-3-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/cluster.Rmd/unnamed-chunk-3-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.heatmap.for.genes(&quot;ETV&quot;, AF.patients)</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-3-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/cluster.Rmd/unnamed-chunk-3-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cluster-af-patients-based-on-tf-activity" class="section level1">
<h1>cluster AF patients based on TF activity</h1>
<pre class="r"><code>data= data.filt

target= data$target
data(dorothea_hs, package = &quot;dorothea&quot;)

regulons &lt;- dorothea_hs %&gt;%
  dplyr::filter(confidence %in% c(&quot;A&quot;, &quot;B&quot;,&quot;C&quot;))

# some ETV TFs are low confidence. we add them manually to the selection
dorothea_hs %&gt;% filter(grepl(&quot;ETV&quot;, tf)) %&gt;% distinct(tf, confidence)</code></pre>
<pre><code># A tibble: 7 x 2
  tf    confidence
  &lt;chr&gt; &lt;chr&gt;     
1 ETV1  C         
2 ETV2  E         
3 ETV3  E         
4 ETV4  A         
5 ETV5  E         
6 ETV6  D         
7 ETV7  E         </code></pre>
<pre class="r"><code>regulons.extended = rbind(regulons, dorothea_hs %&gt;% filter(grepl(&quot;ETV&quot;, tf), 
                                                           !confidence %in% c(&quot;A&quot;, &quot;B&quot;,&quot;C&quot;)) 
) %&gt;% mutate(label= paste0(tf,&quot;_&quot;, confidence)) %&gt;%
  select(-tf)%&gt;% rename(tf= label)
#scale(data$GEX$voom_count,center = T, scale = T)

regulons.extended = rbind(regulons, dorothea_hs %&gt;% filter(grepl(&quot;ETV&quot;, tf), 
                                                           !confidence %in% c(&quot;A&quot;, &quot;B&quot;,&quot;C&quot;)) 
) %&gt;% mutate(likelihood=1)


#### save regulons for inspection:
ETV_regulons=regulons.extended  %&gt;% filter(grepl(&quot;ETV&quot;, tf))
ETV_regilons_list= lapply(unique(ETV_regulons$tf), function(x){ETV_regulons %&gt;% filter(tf == x)})
names(ETV_regilons_list)= unique(ETV_regulons$tf)
library(XML)
WriteXLS::WriteXLS(ETV_regilons_list, ExcelFileName = &quot;output//ETV_regs.xls&quot;)
# now check kcnk3
dorothea_hs %&gt;% filter(target == &quot;KCNK3&quot;) %&gt;% write.csv(., file =&quot;output/KCNK3_potentialTFs.csv&quot;)

##choose mehtod
meth= c(&quot;scira&quot;, &quot;gsva&quot;)

# perform enrichment:
dec.res= decouple(scale(data$GEX$voom_count,center = T, scale = T), 
                  network = regulons.extended,
                  statistics = meth, 
                  .source =tf,
                  .target = target,
                  include_time = F)</code></pre>
<pre><code>Estimating GSVA scores for 297 gene sets.
Estimating ECDFs with Gaussian kernels

  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |                                                                      |   1%
  |                                                                            
  |=                                                                     |   1%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |==                                                                    |   2%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |===                                                                   |   5%
  |                                                                            
  |====                                                                  |   5%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |=====                                                                 |   7%
  |                                                                            
  |=====                                                                 |   8%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |======                                                                |   9%
  |                                                                            
  |=======                                                               |   9%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |=========                                                             |  12%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |==========                                                            |  15%
  |                                                                            
  |===========                                                           |  15%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |============                                                          |  16%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |============                                                          |  18%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |=============                                                         |  19%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |==============                                                        |  21%
  |                                                                            
  |===============                                                       |  21%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |================                                                      |  22%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |================                                                      |  24%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |=================                                                     |  25%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |===================                                                   |  28%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |=====================                                                 |  29%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |=====================                                                 |  31%
  |                                                                            
  |======================                                                |  31%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |=======================                                               |  32%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |========================                                              |  35%
  |                                                                            
  |=========================                                             |  35%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |==========================                                            |  38%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |===========================                                           |  39%
  |                                                                            
  |============================                                          |  39%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |==============================                                        |  42%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |===============================                                       |  45%
  |                                                                            
  |================================                                      |  45%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |=================================                                     |  46%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |=================================                                     |  48%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |===================================                                   |  49%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |===================================                                   |  51%
  |                                                                            
  |====================================                                  |  51%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |=====================================                                 |  52%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |=====================================                                 |  54%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |======================================                                |  55%
  |                                                                            
  |=======================================                               |  55%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |========================================                              |  58%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |==========================================                            |  61%
  |                                                                            
  |===========================================                           |  61%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |============================================                          |  62%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |=============================================                         |  65%
  |                                                                            
  |==============================================                        |  65%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |===============================================                       |  68%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |================================================                      |  69%
  |                                                                            
  |=================================================                     |  69%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |=================================================                     |  71%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |===================================================                   |  72%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |=====================================================                 |  75%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |======================================================                |  76%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |======================================================                |  78%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |=======================================================               |  79%
  |                                                                            
  |========================================================              |  79%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |=========================================================             |  81%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |==========================================================            |  82%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |==========================================================            |  84%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |===========================================================           |  85%
  |                                                                            
  |============================================================          |  85%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |=============================================================         |  88%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |===============================================================       |  91%
  |                                                                            
  |================================================================      |  91%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |=================================================================     |  92%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |==================================================================    |  95%
  |                                                                            
  |===================================================================   |  95%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |====================================================================  |  98%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |===================================================================== |  99%
  |                                                                            
  |======================================================================|  99%
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre class="r"><code># dec.res= decouple(data$GEX$voom_count, network = regulons.extended,statistics =  c(&quot;gsva&quot;, &quot;mean&quot;, &quot;pscira&quot;, &quot;scira&quot;, &quot;viper&quot;), .source =tf, .target = target )

map(c(&quot;gsva&quot;, &quot;scira&quot;)[1], function(x){
dec.res.m = dec.res %&gt;%
  filter(statistic== x, 
         grepl(&quot;AF&quot;, condition))%&gt;%
  pivot_wider(id_cols= c(condition, score, tf), names_from = condition, values_from  = score) %&gt;% 
  as.data.frame() %&gt;%
 column_to_rownames(&quot;tf&quot;)%&gt;% as.matrix

pheatmap(dec.res.m)
pheatmap(dec.res.m[grepl(&quot;ETV&quot;, rownames(dec.res.m)), ])

})</code></pre>
<p><img src="figure/cluster.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/4bfadb4d9fa2630d570681a23dbc542451009603/docs/figure/cluster.Rmd/unnamed-chunk-4-1.png" target="_blank">4bfadb4</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cluster.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JanLanzer/SFB_arrythmia/blob/4bfadb4d9fa2630d570681a23dbc542451009603/docs/figure/cluster.Rmd/unnamed-chunk-4-2.png" target="_blank">4bfadb4</a>
</td>
<td>
JanLanzer
</td>
<td>
2021-07-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[[1]]</code></pre>
<p>Patients 1.28, 1-29, 1.26 and 1.27 show high ETV4, ETV6, and ETV7 activity. Same patients as in fibrosis analysis Careful, some ETV regulons are inferred and might be less accurate. Patient 19_AF is a bit inbetween</p>
</div>
<div id="next-we-will-use-these-two-patient-groups-of-af-to-test-for-clinical-variables." class="section level1">
<h1>Next we will use these two patient groups of AF to test for clinical variables.</h1>
<pre class="r"><code>cluster.fib= c(&quot;1.28_AF&quot;, &quot;1.29_AF&quot;, &quot;1.26_AF&quot;,&quot;1.27_AF&quot;)
target= data.filt$target

target = target %&gt;% filter(group ==&quot;AF&quot;) %&gt;% mutate(cluster= ifelse(sampleID %in% cluster.fib, &quot;fibrotic&quot;, &quot;non.fibrotic&quot;))

### For categorical data, a chi sqr test is performed om the cluster variable: 

fact.target= target[,37:length(colnames(target))] %&gt;% mutate_all(as.factor) %&gt;% select(-cluster)

test.res= map(colnames(fact.target), function(x){
  #print(x)
  if(length(unique(target[[x]]))&gt;1){
    test.res= chisq.test(target[[x]], target$cluster)
    
    #get.it= c(test.res$p.value, test.res$statistic)
    return(test.res)
  }
  return(NULL)
})</code></pre>
<pre><code>Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect

Warning in chisq.test(target[[x]], target$cluster): Chi-squared approximation
may be incorrect</code></pre>
<pre class="r"><code>#test.res[[1]]$statistic
names(test.res) = colnames(fact.target)
test.res= do.call(rbind, test.res) 
test.res= test.res%&gt;%as.data.frame()  %&gt;% arrange(p.value) #%&gt;% print(n=200)
#test.res= fisher.test(x= target$cluster, y= target$CABG)
print(test.res)</code></pre>
<pre><code>                          statistic parameter    p.value
Heartfailure               2.933333         3  0.4020185
Corononaryheartdisease     3.653571         3  0.3013838
Smoker                     4.301786         2  0.1163802
Femalegender              0.1364087         1  0.7118776
AVR                       0.1604167         1  0.6887732
Hyperlipidemia            0.1604167         1  0.6887732
ACEinhibitor              0.1604167         1  0.6887732
CABG                    0.003507653         1  0.9527725
Myocardialinfarction     0.08839286         1  0.7662303
Antiplateletdrug         0.08839286         1  0.7662303
DOAC                     0.08839286         1  0.7662303
PCI                    5.336257e-31         1          1
Ca.antagonist          5.336257e-31         1          1
Type.IIdiabetes           0.3314732         1  0.5647929
Pulmonaryhypertension      2.753274         1 0.09705552
COPD                              0         1          1
AT1antagonist                     0         1          1
\\beta.blocker         1.667916e-31         1          1
Loopdiuretic           1.667916e-31         1          1
Marcumar               1.667916e-31         1          1
Statin                 1.667916e-31         1          1
                                                                             method
Heartfailure                                             Pearson&#39;s Chi-squared test
Corononaryheartdisease                                   Pearson&#39;s Chi-squared test
Smoker                                                   Pearson&#39;s Chi-squared test
Femalegender           Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
AVR                    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
Hyperlipidemia         Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
ACEinhibitor           Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
CABG                   Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
Myocardialinfarction   Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
Antiplateletdrug       Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
DOAC                   Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
PCI                    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
Ca.antagonist          Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
Type.IIdiabetes        Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
Pulmonaryhypertension  Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
COPD                   Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
AT1antagonist          Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
\\beta.blocker         Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
Loopdiuretic           Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
Marcumar               Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
Statin                 Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
                                            data.name               observed
Heartfailure           target[[x]] and target$cluster 0, 0, 3, 1, 2, 1, 2, 2
Corononaryheartdisease target[[x]] and target$cluster 1, 1, 0, 2, 1, 0, 3, 3
Smoker                 target[[x]] and target$cluster       1, 3, 0, 4, 1, 2
Femalegender           target[[x]] and target$cluster             4, 0, 5, 2
AVR                    target[[x]] and target$cluster             1, 3, 4, 3
Hyperlipidemia         target[[x]] and target$cluster             1, 3, 4, 3
ACEinhibitor           target[[x]] and target$cluster             3, 1, 3, 4
CABG                   target[[x]] and target$cluster             2, 2, 2, 5
Myocardialinfarction   target[[x]] and target$cluster             3, 1, 7, 0
Antiplateletdrug       target[[x]] and target$cluster             1, 3, 0, 7
DOAC                   target[[x]] and target$cluster             3, 1, 7, 0
PCI                    target[[x]] and target$cluster             4, 0, 6, 1
Ca.antagonist          target[[x]] and target$cluster             4, 0, 6, 1
Type.IIdiabetes        target[[x]] and target$cluster             2, 2, 6, 1
Pulmonaryhypertension  target[[x]] and target$cluster             0, 4, 5, 2
COPD                   target[[x]] and target$cluster             3, 1, 5, 2
AT1antagonist          target[[x]] and target$cluster             3, 1, 5, 2
\\beta.blocker         target[[x]] and target$cluster             1, 3, 1, 6
Loopdiuretic           target[[x]] and target$cluster             1, 3, 1, 6
Marcumar               target[[x]] and target$cluster             3, 1, 6, 1
Statin                 target[[x]] and target$cluster             1, 3, 1, 6
                                                                                                     expected
Heartfailure           0.7272727, 0.3636364, 1.8181818, 1.0909091, 1.2727273, 0.6363636, 3.1818182, 1.9090909
Corononaryheartdisease 0.7272727, 0.3636364, 1.0909091, 1.8181818, 1.2727273, 0.6363636, 1.9090909, 3.1818182
Smoker                                       1.8181818, 1.4545455, 0.7272727, 3.1818182, 2.5454545, 1.2727273
Femalegender                                                       3.2727273, 0.7272727, 5.7272727, 1.2727273
AVR                                                                    1.818182, 2.181818, 3.181818, 3.818182
Hyperlipidemia                                                         1.818182, 2.181818, 3.181818, 3.818182
ACEinhibitor                                                           2.181818, 1.818182, 3.818182, 3.181818
CABG                                                                   1.454545, 2.545455, 2.545455, 4.454545
Myocardialinfarction                                               3.6363636, 0.3636364, 6.3636364, 0.6363636
Antiplateletdrug                                                   0.3636364, 3.6363636, 0.6363636, 6.3636364
DOAC                                                               3.6363636, 0.3636364, 6.3636364, 0.6363636
PCI                                                                3.6363636, 0.3636364, 6.3636364, 0.6363636
Ca.antagonist                                                      3.6363636, 0.3636364, 6.3636364, 0.6363636
Type.IIdiabetes                                                        2.909091, 1.090909, 5.090909, 1.909091
Pulmonaryhypertension                                                  1.818182, 2.181818, 3.181818, 3.818182
COPD                                                                   2.909091, 1.090909, 5.090909, 1.909091
AT1antagonist                                                          2.909091, 1.090909, 5.090909, 1.909091
\\beta.blocker                                                     0.7272727, 3.2727273, 1.2727273, 5.7272727
Loopdiuretic                                                       0.7272727, 3.2727273, 1.2727273, 5.7272727
Marcumar                                                           3.2727273, 0.7272727, 5.7272727, 1.2727273
Statin                                                             0.7272727, 3.2727273, 1.2727273, 5.7272727
                                                                                                                residuals
Heartfailure           -0.85280287, -0.60302269, 0.87645982, -0.08703883, 0.64465837, 0.45584231, -0.66254135, 0.06579517
Corononaryheartdisease         0.3198011, 1.0552897, -1.0444659, 0.1348400, -0.2417469, -0.7977240, 0.7895420, -0.1019294
Smoker                                                -0.6067799, 1.2814232, -0.8528029, 0.4586825, -0.9686649, 0.6446584
Femalegender                                                                 0.4020151, -0.8528029, -0.3038949, 0.6446584
AVR                                                                          -0.6067799, 0.5539117, 0.4586825, -0.4187179
Hyperlipidemia                                                               -0.6067799, 0.5539117, 0.4586825, -0.4187179
ACEinhibitor                                                                 0.5539117, -0.6067799, -0.4187179, 0.4586825
CABG                                                                         0.4522670, -0.3418817, -0.3418817, 0.2584383
Myocardialinfarction                                                         -0.3337119, 1.0552897, 0.2522625, -0.7977240
Antiplateletdrug                                                             1.0552897, -0.3337119, -0.7977240, 0.2522625
DOAC                                                                         -0.3337119, 1.0552897, 0.2522625, -0.7977240
PCI                                                                          0.1906925, -0.6030227, -0.1441500, 0.4558423
Ca.antagonist                                                                0.1906925, -0.6030227, -0.1441500, 0.4558423
Type.IIdiabetes                                                              -0.5330018, 0.8703883, 0.4029115, -0.6579517
Pulmonaryhypertension                                                        -1.3483997, 1.2309149, 1.0192944, -0.9304842
COPD                                                                     0.05330018, -0.08703883, -0.04029115, 0.06579517
AT1antagonist                                                            0.05330018, -0.08703883, -0.04029115, 0.06579517
\\beta.blocker                                                               0.3198011, -0.1507557, -0.2417469, 0.1139606
Loopdiuretic                                                                 0.3198011, -0.1507557, -0.2417469, 0.1139606
Marcumar                                                                     -0.1507557, 0.3198011, 0.1139606, -0.2417469
Statin                                                                       0.3198011, -0.1507557, -0.2417469, 0.1139606
                                                                                                           stdres
Heartfailure           -1.1818737, -0.7928250, 1.4876476, -0.1279416, 1.1818737, 0.7928250, -1.4876476, 0.1279416
Corononaryheartdisease 0.4432026, 1.3874437, -1.5352989, 0.2288689, -0.4432026, -1.3874437, 1.5352989, -0.2288689
Smoker                                              -1.029910, 2.013665, -1.181874, 1.029910, -2.013665, 1.181874
Femalegender                                                             1.181874, -1.181874, -1.181874, 1.181874
AVR                                                                          -1.02991, 1.02991, 1.02991, -1.02991
Hyperlipidemia                                                               -1.02991, 1.02991, 1.02991, -1.02991
ACEinhibitor                                                                 1.02991, -1.02991, -1.02991, 1.02991
CABG                                                                 0.7107053, -0.7107053, -0.7107053, 0.7107053
Myocardialinfarction                                                     -1.387444, 1.387444, 1.387444, -1.387444
Antiplateletdrug                                                         1.387444, -1.387444, -1.387444, 1.387444
DOAC                                                                     -1.387444, 1.387444, 1.387444, -1.387444
PCI                                                                      0.792825, -0.792825, -0.792825, 0.792825
Ca.antagonist                                                            0.792825, -0.792825, -0.792825, 0.792825
Type.IIdiabetes                                                          -1.279416, 1.279416, 1.279416, -1.279416
Pulmonaryhypertension                                                    -2.288689, 2.288689, 2.288689, -2.288689
COPD                                                                 0.1279416, -0.1279416, -0.1279416, 0.1279416
AT1antagonist                                                        0.1279416, -0.1279416, -0.1279416, 0.1279416
\\beta.blocker                                                       0.4432026, -0.4432026, -0.4432026, 0.4432026
Loopdiuretic                                                         0.4432026, -0.4432026, -0.4432026, 0.4432026
Marcumar                                                             -0.4432026, 0.4432026, 0.4432026, -0.4432026
Statin                                                               0.4432026, -0.4432026, -0.4432026, 0.4432026</code></pre>
<pre class="r"><code>### now run t.test for the continous variables
cont.target= target[,1:35] %&gt;% select(-sampleID)

ttest.res= map(colnames(cont.target), function(x){
  if( is.numeric(target[[x]])){
    fib.samples= target %&gt;% filter(sampleID %in% cluster.fib)
    n.fib=  target %&gt;% filter(!sampleID %in% cluster.fib)
    if(sum(!is.na(fib.samples[[x]]))&gt;2 &amp; sum(!is.na(n.fib[[x]]))&gt;2){
      test.res= t.test( n.fib[[x]], fib.samples[[x]])
      return(c(test.res$statistic, test.res$p.value))
      
    }
    
  }
  return(NULL)
})
names(ttest.res) = colnames(cont.target)
ttest.res= do.call(rbind, ttest.res) 
colnames(ttest.res)= c(&quot;t&quot;, &quot;p.val&quot;)
ttest.res= ttest.res%&gt;%as.data.frame()  %&gt;% rownames_to_column(&quot;feature&quot;) %&gt;% as_tibble()  %&gt;%  mutate(logp= -log10(p.val))
ttest.res %&gt;% arrange(desc(logp)) %&gt;% print(n=200)</code></pre>
<pre><code># A tibble: 27 x 4
   feature                   t  p.val   logp
   &lt;chr&gt;                 &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 ALT/GPT             -2.13   0.0639 1.19  
 2 MCH                 -1.86   0.105  0.977 
 3 Alkalinephosphatase  1.74   0.131  0.884 
 4 HypochromousRBC      1.50   0.167  0.777 
 5 LDH                 -1.50   0.174  0.759 
 6 MCV                 -1.34   0.255  0.594 
 7 Creatinine           0.853  0.423  0.374 
 8 CRP                  0.837  0.425  0.371 
 9 Leukocytes           0.838  0.440  0.356 
10 Height              -0.819  0.441  0.356 
11 Age                  0.791  0.449  0.348 
12 Hematocrit           0.775  0.458  0.339 
13 AST/GOT             -0.804  0.468  0.330 
14 Bilirubintotal       0.618  0.553  0.258 
15 BMI                 -0.601  0.563  0.249 
16 GFR(CKD.E)          -0.599  0.564  0.248 
17 RDW.CV              -0.573  0.587  0.231 
18 TroponinT            0.565  0.589  0.230 
19 Hemoglobin           0.537  0.605  0.218 
20 RBC                  0.528  0.612  0.213 
21 Pancreaticamylase    0.461  0.656  0.183 
22 GGT                 -0.431  0.677  0.169 
23 MDRD                -0.387  0.709  0.150 
24 Weight              -0.270  0.794  0.100 
25 Urea                -0.243  0.814  0.0896
26 MCHC                 0.232  0.822  0.0853
27 Lipase              -0.0990 0.928  0.0325</code></pre>
<p>with the proposed tow patient groups, there seems to be no association with meaningful clinical vars. Sample size is probably too limiting</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Linux Mint 19.1

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] XML_3.99-0.6     lsa_0.73.2       SnowballC_0.7.0  pheatmap_1.0.12 
 [5] cowplot_1.1.1    decoupleR_0.99.2 dorothea_1.0.1   forcats_0.5.1   
 [9] stringr_1.4.0    dplyr_1.0.6      purrr_0.3.4      readr_1.4.0     
[13] tidyr_1.1.3      tibble_3.1.1     ggplot2_3.3.5    tidyverse_1.3.1 
[17] workflowr_1.6.2 

loaded via a namespace (and not attached):
  [1] colorspace_2.0-1            ellipsis_0.3.2             
  [3] rprojroot_2.0.2             XVector_0.30.0             
  [5] GenomicRanges_1.42.0        fs_1.5.0                   
  [7] rstudioapi_0.13             farver_2.1.0               
  [9] bit64_4.0.5                 AnnotationDbi_1.50.3       
 [11] fansi_0.4.2                 lubridate_1.7.10           
 [13] xml2_1.3.2                  cachem_1.0.4               
 [15] knitr_1.33                  jsonlite_1.7.2             
 [17] speedglm_0.3-3              bcellViper_1.24.0          
 [19] broom_0.7.6                 annotate_1.66.0            
 [21] dbplyr_2.1.1                graph_1.66.0               
 [23] compiler_4.0.2              httr_1.4.2                 
 [25] backports_1.2.1             assertthat_0.2.1           
 [27] Matrix_1.3-3                fastmap_1.1.0              
 [29] cli_2.5.0                   later_1.2.0                
 [31] htmltools_0.5.1.1           tools_4.0.2                
 [33] gtable_0.3.0                glue_1.4.2                 
 [35] GenomeInfoDbData_1.2.4      Rcpp_1.0.6                 
 [37] Biobase_2.50.0              cellranger_1.1.0           
 [39] jquerylib_0.1.4             vctrs_0.3.8                
 [41] xfun_0.22                   rvest_1.0.0                
 [43] lifecycle_1.0.0             WriteXLS_6.3.0             
 [45] zlibbioc_1.36.0             MASS_7.3-54                
 [47] scales_1.1.1                hms_1.0.0                  
 [49] promises_1.2.0.1            MatrixGenerics_1.2.1       
 [51] parallel_4.0.2              SummarizedExperiment_1.20.0
 [53] RColorBrewer_1.1-2          yaml_2.2.1                 
 [55] memoise_2.0.0               sass_0.4.0                 
 [57] stringi_1.6.1               RSQLite_2.2.7              
 [59] GSVA_1.38.2                 highr_0.9                  
 [61] S4Vectors_0.28.1            BiocGenerics_0.36.1        
 [63] BiocParallel_1.24.1         GenomeInfoDb_1.26.7        
 [65] rlang_0.4.11                pkgconfig_2.0.3            
 [67] bitops_1.0-7                matrixStats_0.58.0         
 [69] evaluate_0.14               lattice_0.20-44            
 [71] labeling_0.4.2              bit_4.0.4                  
 [73] tidyselect_1.1.1            GSEABase_1.50.1            
 [75] magrittr_2.0.1              R6_2.5.0                   
 [77] IRanges_2.24.1              generics_0.1.0             
 [79] DelayedArray_0.16.3         DBI_1.1.1                  
 [81] pillar_1.6.0                haven_2.4.1                
 [83] whisker_0.4                 withr_2.4.2                
 [85] RCurl_1.98-1.3              modelr_0.1.8               
 [87] crayon_1.4.1                utf8_1.2.1                 
 [89] rmarkdown_2.8               grid_4.0.2                 
 [91] readxl_1.3.1                blob_1.2.1                 
 [93] git2r_0.28.0                reprex_2.0.0               
 [95] digest_0.6.27               xtable_1.8-4               
 [97] httpuv_1.6.1                stats4_4.0.2               
 [99] munsell_0.5.0               bslib_0.2.4                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
